src/mesh/mesh_tools.C

```diff
@@ -1128,16 +1128,23 @@ void MeshTools::libmesh_assert_valid_remote_elems(const MeshBase & mesh)
     {
       const Elem * elem = *el;
       libmesh_assert (elem);
-      for (unsigned int n=0; n != elem->n_neighbors(); ++n)
-        libmesh_assert_not_equal_to (elem->neighbor_ptr(n), remote_elem);
+
+      // We currently don't allow active_local_elements to have
+      // remote_elem neighbors
+      if (elem->active())
+        for (unsigned int n=0; n != elem->n_neighbors(); ++n)
+          libmesh_assert_not_equal_to (elem->neighbor_ptr(n), remote_elem);
```

Comment:
Wait.  That whole function is in our "crazy expensive verification tests" MeshTools suite, which isn't even defined unless ifdef DEBUG.

Still, good lesson for future optimization.  For loops aren't always as cheap as they look or as they should be.